pal_event_server/lib/src/loggers/app_usage/macos/macos_helper.dart

```diff
@@ -18,39 +18,53 @@ import 'dart:async';
 import 'dart:io';
 import 'dart:isolate';
 
+import '../idle_time.dart';
+
 import '../../pal_event_helper.dart';
 import '../app_logger.dart';
 import 'apple_script_util.dart' as apple_script;
 
 String _prevAppAndWindowName;
+bool _prevIdleState;
 
 // Isolate entry point must be a top-level function (or static?)
 // Run Apple Script for the active window
 void macOSAppLoggerIsolate(SendPort sendPort) {
   Timer.periodic(queryInterval, (Timer _) {
     try {
-      Process.run(apple_script.command, apple_script.scriptArgs).then((result) {
-        final currWindow = result.stdout.trim();
+      Future.wait([
+        Process.run(apple_script.command, apple_script.scriptArgs),
+        isIdle()
+      ]).then((result) {
+        final currWindow = (result[0] as ProcessResult).stdout.trim();
+        final idleState = result[1];
```

Comment:
is it possible to get an out of bounds error here?

Reply:
Same as above

Reply:
Thanks